### PR TITLE
feat: Use filesystem specific checks for host preflights

### PR DESCRIPTION
**Pull request by hedge-sparrow**
sc-107808

### CODE REVIEW

**Comment by hedge-sparrow** [comment]
image

File: pkg/preflights/host-preflight.yaml [hunk 1]
```diff
@@ -22,20 +34,56 @@ spec:
...
+              message: The filesystem at the path /opt/openebs is more than 80% full
+          - pass:
+              message: The filesystem at the path /opt/openebs has sufficient space
+    - diskUsage:
+        checkName: tmp path usage
+        collectorName: tmp-path-usage
+        outcomes:
+          - fail:
+              when: 'total < 5Gi'
+              message: The filesystem at the path /tmp has less than 40Gi of total space
```
Comment (ajp-io) [inline pkg/preflights/host-preflight.yaml:81]:
```suggestion
              message: The filesystem at the path /tmp has less than 5Gi of total space
```

File: pkg/preflights/host-preflight.yaml [hunk 2]
```diff
@@ -22,20 +34,56 @@ spec:
...
+              message: The filesystem at the path /var/lib/k0s is more than 80% full
+          - pass:
+              message: The filesystem at the path /var/lib/k0s has sufficient space
+    - diskUsage:
+        checkName: openebs path usage
+        collectorName: openebs-path-usage
+        outcomes:
+          - fail:
+              when: 'total < 5Gi'
+              message: The filesystem at the path /opt/openebs has less than 40Gi of total space
```
Comment (ajp-io) [inline pkg/preflights/host-preflight.yaml:69]:
```suggestion
              message: The filesystem at the path /opt/openebs has less than 5Gi of total space
```

**Review by ajp-io — COMMENTED** [review]
Another thing I discussed with Chris is that the used/total checks don't seem helpful in a few cases. Like for /tmp, if you have the needed 5Gi, you're fine. Checking the percentage doesn't add a lot more value, and it might even cause a problem if someone has a huge disk that is 90% full. The preflight will fail, but they could have plenty of space.

Chris mentioned that Kubernetes does start garbage collecting if a disk is 85% full. So the proposal is to remove the "percent full" checks, except for whichever one Kubernetes cares about. That's presumably /var/lib/k0s, but we should be sure.

**Review by ajp-io — APPROVED** [review]
lgtm

**Pull request merged**